registry/config.go

```diff
@@ -223,8 +223,8 @@ func validateRemoteName(remoteName string) error {
 	if !validNamespaceChars.MatchString(namespace) {
 		return fmt.Errorf("Invalid namespace name (%s). Only [a-z0-9-_] are allowed.", namespace)
 	}
-	if len(namespace) < 4 || len(namespace) > 30 {
-		return fmt.Errorf("Invalid namespace name (%s). Cannot be fewer than 4 or more than 30 characters.", namespace)
+	if len(namespace) < 4 || len(namespace) > 256 {
```

Comment:
The v2 registry currently has a minimum of 2, not 4: https://github.com/docker/distribution/blob/master/registry/api/v2/names.go#L16